packages/dids/src/methods/did-key.ts

```diff
@@ -345,83 +343,44 @@ export class DidKey extends DidMethod {
     keyManager?: TKms;
     options?: DidKeyCreateOptions<TKms>;
   } = {}): Promise<BearerDid> {
+    // Before processing the create operation, validate DID-method-specific requirements to prevent
+    // keys from being generated unnecessarily.
+
+    // Check 1: Validate that `algorithm` or `verificationMethods` options are not both given.
     if (options.algorithm && options.verificationMethods) {
       throw new Error(`The 'algorithm' and 'verificationMethods' options are mutually exclusive`);
     }
 
+    // Check 2: If `verificationMethods` is given, it must contain exactly one entry since DID JWK
+    // only supports a single verification method.
+    if (options.verificationMethods && options.verificationMethods.length !== 1) {
+      throw new Error(`The 'verificationMethods' option must contain exactly one entry`);
+    }
+
     // Default to Ed25519 key generation if an algorithm is not given.
     const algorithm = options.algorithm ?? options.verificationMethods?.[0]?.algorithm ?? 'Ed25519';
 
     // Generate a new key using the specified `algorithm`.
     const keyUri = await keyManager.generateKey({ algorithm });
     const publicKey = await keyManager.getPublicKey({ keyUri });
 
-    // Create the DID object from the generated key material, including DID document, metadata,
-    // signer convenience function, and URI.
-    const did = await DidKey.fromPublicKey({ keyManager, publicKey, options });
+    // Compute the DID identifier from the public key by converting the JWK to a multibase-encoded
```

Comment:
Is this comment valid ?

Reply:
Sure is -- that's how DID identifiers are encoded for DID Key